graphql_metrics.gemspec

```diff
@@ -33,16 +33,19 @@ Gem::Specification.new do |spec|
   spec.executables   = spec.files.grep(%r{^exe/}) { |f| File.basename(f) }
   spec.require_paths = ["lib"]
 
+  spec.add_runtime_dependency "concurrent-ruby", "~> 1.1.0"
```

Comment:
?

Reply:
This is explained here: https://github.com/Shopify/graphql-metrics/pull/10/files#diff-eb87a16ba8d7a5b04c0484ae1f802b83R5-R7

I suppose I could use `Thread.current[]=`? I figured the concurreny-ruby gem is preferable due to its maturity.

Is it silly to worry about the same tracer running concurrently?. I'm not sure if multiplexing in the gem can / will support doing so across multiple queries at once in the future. It seemed safer to do this, to keep values from the lexing, parsing phase before we have a context to store values in for later.

Reply:
Confirmed, this library ultimately uses `Thread.current[]=` (just like tracers for graphql-ruby for Skylight, NewRelic), but with better guarantees (and some slowdown, hopefully negligible) due to locking w/ a Mutex.

https://github.com/ruby-concurrency/concurrent-ruby/blob/master/lib/concurrent/atomic/ruby_thread_local_var.rb